### ISSUE TITLE
Spurious notifications and weird behavior when going to the home app

### ISSUE THREAD

**Issue by patricegautier** [hjdhjd/homebridge-myq2#92]
I am now seeing the following behavior fairly consistently on two different home bridges, both up-to-date with the latest Home bridge and myq2 (2.1.3) plug-in:

Upon going to the home app I get a Closing… State then the state switches to closed and I get a notification that the door was opened and then another one the door was closed.  Of course I have done nothing at all to the door..

In the logs (debug on):

[7/18/2020, 7:33:03 PM] [myQ] Can't find device: MyQ 99 Garage Door - 90b2230c-5b7d-415b-b6d9-xyz.
[7/18/2020, 7:33:03 PM] [myQ] Can't find device: MyQ 99 Garage Door - 90b2230c-5b7d-415b-b6d9-xyz.
[7/18/2020, 7:33:38 PM] [myQ] {
  count: 8,
  href: 'http://api.myqdevice.com/api/v5/accounts/fd5df018-be38-4dcc-8ded-7dbff51b7ebd/devices',
  items: [
    {
      created_date: '2019-09-23T16:09:22.763',
      device_family: 'garagedoor',
  ...

Then the home bridge behaves correctly for a little while

**Comment by hjdhjd** [comment]
@patricegautier The "Closing..." is a Home app artifact that primarily occurs on a combination of homebridge startup and starting up the Home app for the first time in a while and it tries to refresh it's own status. Not terribly concerned about that one.

What is interesting is the can't find device errors. Alright, let's start with - what's your actual physical setup for your myQ devices...it sounds like you've got several devices connected to several bridges?

Another is, out of an abundance of caution, you might want to try to clear out your accessories cache, in case something's been corrupted along the way.

Thanks.

**Comment by patricegautier** [comment]
> @patricegautier The "Closing..." is a Home app artifact that primarily occurs on a combination of homebridge startup and starting up the Home app for the first time in a while and it tries to refresh it's own status. Not terribly concerned about that one
> 

Ok.. It is definitely new Since 2.1 though And seems to happen beyond start up of home bridge. I will confirm that though..

> What is interesting is the can't find device errors. Alright, let's start with - what's your actual physical setup for your myQ devices...it sounds like you've got several devices connected to several bridges?
> 

Two separate home Bridges in 2 different homes: one with two door openers connected directly to home bridge, and another one with one opener connected to a bridge connected to the home bridge. The error occurs on both

> Another is, out of an abundance of caution, you might want to try to clear out your accessories cache, in case something's been corrupted along the way.
> 

I did that on one of those bridges cannot remember which one and had to re-set up a bunch of things beyond MyQ so I’m a little leery of doing it again :(

Thank you.

**Comment by hjdhjd** [comment]
Closing... is not new. I've seen it for months. :smile: But if you're seeing something that's truly an **issue** then let's troubleshoot it. If you're reporting something cosmetic that disappears shortly after startup...that's not going to the top of the queue.

When you say home bridges...are you referring to the home bridge hardware device that Liftmaster/Chamberlain sells, or an instance of homebridge, the project that this is a plugin for? Sorry, it gets confusing.

The errors you're describing are related to the plugin either thinking accessories are there that aren't, or that the myQ API is somehow making those devices appear and disappear. I haven't seen the API exhibit that behavior yet, but hey...it's an undocumented and very finicky API, so who knows...

But if it's the former problem...you can definitely start with checking the accessory cache. There's a way to look at the accessory cache and selectively delete things there if you use the homebridge UI. Go to homebridge settings | Unpair bridges...and delete the entries related to myQ devices or others you want to get rid of.

**Comment by patricegautier** [comment]
OK I found the way to selectively remove excess weight from the cash and removed all the myQ2 entries

> When you say home bridges...are you referring to the home bridge hardware device that Liftmaster/Chamberlain sells, or an instance of homebridge, the project that this is a plugin for? Sorry, it gets confusing.
> 
Let me rephrase:

Two separate raspberry pies home Bridges in 2 different homes: one with two door openers connected directly to the pi home bridge, and another one with one opener connected to a chamberlain bridge connected to the pi home bridge.

I will report..

**Comment by hjdhjd** [comment]
Patrice...I was able to recreate the bug you mentioned. I'm working on it.

**Comment by hjdhjd** [comment]
A fix for the "Closing..." issue will be in the next release. Thanks!

**Comment by patricegautier** [comment]
No thank you! And thanks for your work on this plug-in…

**Comment by hjdhjd** [comment]
@patricegautier Please try 2.1.4. I believe the "Closing" issue you mentioned should be addressed. Thanks.

**Comment by patricegautier** [comment]
Installing the update.. will let you know!

**Comment by hjdhjd** [comment]
Look forward to it...

**Comment by patricegautier** [comment]
It’s only been a few hours, but I can already tell you the bug is fixed.. I haven’t seen any occurrence of the issue and it was really easy to reproduce.

**Comment by hjdhjd** [comment]
Great! Closing this one out. Thanks for reporting.
